### PR TITLE
Deprecate errors field

### PR DESCRIPTION
**Pull request by maarcingebala**
Deprecate the `errors` field in API mutations' responses in favor of typed errors with error codes.

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [x] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/translations/mutations.py [hunk 1]
```diff
@@ -24,6 +26,11 @@ def check_permissions(cls, context):
 
     @classmethod
     def perform_mutation(cls, _root, info, **data):
+        if "id" in data and not data["id"]:
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/mutations.py:29]: `if not data.get("id"):`?

File: saleor/graphql/translations/mutations.py [hunk 2]
```diff
@@ -24,6 +26,11 @@ def check_permissions(cls, context):
...
+            raise ValidationError(
+                {"id": ValidationError("This field is required", code="required")}
+            )
+
```
Comment (IKarbowiak) [inline saleor/graphql/translations/mutations.py:33]: I am curious, `required=True` on the input field is not enough?

Comment (maarcingebala) [reply saleor/graphql/translations/mutations.py:33]: It still allows you to pass `id="` which fails as the `instance` is None then.

File: saleor/graphql/translations/mutations.py (hunk 1 above)
Comment (NyanKiyoshi) [reply saleor/graphql/translations/mutations.py:29]: Comparison: https://gist.github.com/NyanKiyoshi/b221de2be1b0cf2ea89d3de5bf044809

```

In [10]:

data1 = {"id": "abcde", "data": []}
data2 = {"data": []}

In [11]:

%timeit "id" in data1 and not data1["id"]

62.5 ns ± 0.57 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [12]:

%timeit not data1.get("id")

57.6 ns ± 0.674 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [13]:

%timeit "id" in data2 and not data2["id"]

29.3 ns ± 0.286 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [14]:

%timeit not data2.get("id")

54.8 ns ± 0.795 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

```

**Pull request merged**